### PR TITLE
Implement legacy naming for old color palette

### PR DESCRIPTION
**Pull request by dancormier**
> **Note**
> This PR is targeted to merge into `stacks-next`, our long-lived branch intended for our v2 release

**tl;dr**: This PR removes the `new` prefix from all new colors, adds the `legacy` prefix to old colors, refines much of the theme generation, and updates docs. It builds from the previous related PR: https://github.com/StackExchange/Stacks/pull/1415

### Color naming

- New colors no longer include the `new` suffix
- All old colors now include the `legacy` suffix
- All references to legacy, new colors updated
- Internally, legacy and new colors are now referred to as `v1` and `v2` respectively

### Docs

- New docs pages have dropped the `(new)` label
- Legacy docs pages have gained the `(legacy)` label and have been moved to `v1/` path
- New utility color classes (`.fc-*`, `.bc-*`, `.bg-*`) are now entirely documented on the [colors docs page](https://deploy-preview-1440--stacks.netlify.app/product/base/colors/#classes)

### Other

- Moved new color files out of `new` directories, moved legacy into `v1` directories
- Classes previously tied to legacy color generation (such as `*-transparent`) have been moved to new color generation
- Resolved any issues with color generation

## The most relevant changes

> **Note**
> The only changes to all files within `lib/components` should be updating references to legacy colors

Since this PR touches so many files, I figured it may be helpful to highlight files with the most relevant and significant changes:

- [lib/atomic/color.less](https://github.com/StackExchange/Stacks/pull/1440/files#diff-2c75d563f3eae108a7163233a5135d98be0572d91e16b444bfc9cf88435f5b40)
    - [lib/atomic/__snapshots__/color.less.test.ts.snap](https://github.com/StackExchange/Stacks/pull/1440/files#diff-ece4f4e3bf606b04a253b8ec80107a9241127fd1cba05920ce9ba20bcc2f6a54)
- [lib/exports/color.less](https://github.com/StackExchange/Stacks/pull/1440/files#diff-a3ba1b63e7acab9bc351c8b6a6298f4fecd1f04a5732e013fbdf992364764ddc)
    - [lib/exports/__snapshots__/color.less.test.ts.snap](https://github.com/StackExchange/Stacks/pull/1440/files#diff-b5298c2cb59175aea95b33e42b4aa0df9a39bc68d001e7387ea8f625b812f6e2)
- [lib/exports/color-mixins.less](https://github.com/StackExchange/Stacks/pull/1440/files#diff-2bb1aec0e713247f0e67d907306fc440140f214601199c9f65cb70cffc34557f)
- [lib/exports/color-sets.less](https://github.com/StackExchange/Stacks/pull/1440/files#diff-ea72389fc03dddb312af7fc98958fa06757fdf175252829b9468c4a187e6f9a4)
- [lib/exports/v1/constants-colors.less](https://github.com/StackExchange/Stacks/pull/1440/files#diff-3a73fbd25ba9cae90ffc24f7dbcd6b1c9be0133609f09f589f5d4bace3f6dbe4)
    - [lib/exports/v1/__snapshots__/constants-colors.less.test.ts.snap](https://github.com/StackExchange/Stacks/pull/1440/files#diff-33fc59351c7727ff4e1b3129dda309768b8fd0f7a4f9c4ea90b6ea1165dd87db)

## Removing the `new` suffix to new colors

All new colors no longer include the `new` suffix

```Less
/* New */
--white: …;
--black-200: …;
--theme-base-primary-color-h: …;
.fc-theme-primary-400 { … }
```

## Adding the `legacy` suffix to old colors

All old colors now include the `legacy` suffix. References to these colors have been updated.

```Less
/* Legacy */
--white-legacy: …;
--black-legacy-200: …;
--theme-base-primary-color-legacy-h: …;
.fc-theme-primary-legacy-400 { … }
```

## New colors

New colors should match [the Figma](https://www.figma.com/file/cCU6A56sHwZluraDoyg5o7/Colors%3A-Accessible-Color-Palette?type=design&node-id=1303-30963&mode=design&t=IitSCpq8f7TzKmwD-0). See the new [Colors docs page](https://deploy-preview-1415--stacks.netlify.app/product/base/colors-new/) for more details.

### CODE REVIEW

File: adrs/0001-pseudo-private-custom-properties.md [hunk 1]
```diff
@@ -37,23 +37,23 @@ Defining a structure for component styles using pseudo-private custom properties
     @cn-transition-duration: 100ms;
     // BASE COMPONENT-SPECIFIC CUSTOM PROPERTIES
     // --_{component-abbreviation}-{property-shorthand}: {value};
-    --_cn-bg: var(--blue-500);
```
Comment (giamir) [inline adrs/0001-pseudo-private-custom-properties.md:40]: Could we simply update this ADR example with the usage of new colors (and skip adding legacy all together)?

File: adrs/0001-pseudo-private-custom-properties.md [hunk 2]
```diff
@@ -100,7 +100,7 @@ Defining a structure for component styles using pseudo-private custom properties
 
     // INTERACTION
     &:hover {
-        --_cn-bg: var(--_black-050);
+        --_cn-bg: var(--_black-legacy-050);
```
Comment (giamir) [inline adrs/0001-pseudo-private-custom-properties.md]: Was this a typo in the example or we actually support a `var(--_black-050)`? see `_`

File: lib/exports/color-mixins.less [hunk 3]
```diff
@@ -87,41 +89,65 @@
...
+            }
+        }
     }
 
     // Create dark mode atomic class assignments
-    .dark-mode({
-        .d\:bg-@{name} {
-            background-color: @value;
+    & when (@modeSimple = dark) {
+        // Note: we currently do not generate conditional border-color classes
```
Comment (giamir) [inline lib/exports/color-mixins.less:129]: Interesting catch 🤔

File: lib/exports/color-mixins.less [hunk 4]
```diff
@@ -87,41 +89,65 @@
...
  */
-.create-color-classes(@name, @value) {
-    // Create bg atomic class assignments
-    .bc-@{name},
-    .h\:bc-@{name}:hover,
-    .f\:bc-@{name}:focus,
-    .f\:bc-@{name}:focus-within {
-        border-color: @value;
-    }
+.create-color-classes(@name, @value, @modeSimple: light, @property: all) {
```
Comment (giamir) [inline lib/exports/color-mixins.less:94]: Just a thought: it might be beneficial adding a "test" variant with a specific property (instead of the default all).
https://github.com/StackExchange/Stacks/blob/dcormier/legacy-color-rename/lib/exports/color-mixins.less.test.ts#L66-L70

I will leave it up to you to decide, I am also ok to don't test that.

File: lib/exports/color-mixins.less [hunk 5]
```diff
@@ -130,27 +156,22 @@
...
-                @{customVariable}: @value;
-            }
-
-            & when (@type = classes) {
-                .@{name} {
-                    @{property}: @customValue;
-                }
-            }
-        });
+.create-aliased-utility-classes(@modeSimple: light) {
```
Comment (giamir) [inline lib/exports/color-mixins.less:160]: This mixin also does not have a corresponding snapshot. It might be worth adding one. Should go quick.

File: docs/product/base/borders.html [hunk 6]
```diff
@@ -157,330 +150,21 @@
...
-        <div class="stacks-preview--example">
-            <div class="d-flex fd-column w100 gy12">
-                <div class="flex--item ai-center p8 ba bc-silver-darker">Silver darker</div>
-                <div class="flex--item ai-center p8 ba bc-silver">Silver</div>
-                <div class="flex--item ai-center p8 ba bc-silver-lighter">Silver lighter</div>
-            </div>
-        </div>
-    </div>
+    <p class="stacks-copy">
+        Each legacy color stop is available as a border class. See the <a href="{{ "/product/base/colors-legacy#classes" | url }}">colors (legacy) documentation</a> for all available classes.
```
Comment (giamir) [inline docs/product/base/borders.html]: NIT: broken url - should use `product/v1/base/colors`

**Review by giamir — COMMENTED** [review]
Github just lost my PR review summary. Need to retype everything now. 🙃

tldr: looks good, I left few comments but nothing major. Great work 🎉

One note about migration guide:
How did you find all the occurrence that needed to obtain the `legacy` suffix across the codebase? CMD+F on all atomic classes and css variables?
I think your approach might be worth documenting in a migration guide draft since in Core they will need to go through the exact same process most likely.

File: adrs/0001-pseudo-private-custom-properties.md (hunk 1 above)
Comment (dancormier) [reply adrs/0001-pseudo-private-custom-properties.md:40]: I figured I'd change _every_ color reference, including ones here, since it would make a find and replace easy in my follow-up branch. I don't expect anyone to reference this ADR in this state, so figured it's fine for the sake of consistency.

File: adrs/0001-pseudo-private-custom-properties.md (hunk 2 above)
Comment (dancormier) [reply adrs/0001-pseudo-private-custom-properties.md]: That's a typo! Nice catch! I'm on it

File: lib/exports/color-mixins.less (hunk 4 above)
Comment (dancormier) [reply lib/exports/color-mixins.less:94]: I think it's nice to have the additional test variant and it'll be easy to add. Adding it!

**Comment by dancormier** [comment]
> Github just lost my PR review summary. Need to retype everything now. 🙃

Ugh! That's the worst. I've had my prayers to the local storage gods go unheeded before. Feels bad!

> One note about migration guide: How did you find all the occurrence that needed to obtain the `legacy` suffix across the codebase? CMD+F on all atomic classes and css variables? I think your approach might be worth documenting in a migration guide draft since in Core they will need to go through the exact same process most likely.

Good call. I used a few regexes and kinda just went file-by-file on what they found. I figure creating the migration guide will take a little thoughtfulness, so I've made [a ticket](https://stackoverflow.atlassian.net/browse/STACKS-377) for now. IMO, that ticket can be a good holding place for any ideas or tips we want to jot down without the overhead of adding it to the README.

**Comment by dancormier** [comment]
I noticed one minor issue: theme color values are not being overridden in high contrast mode.

image

We should expect the theme colors to match orange/blue for primary/secondary when in high contrast mode, regardless of custom theming. It should be an easy fix. I'll add it into this branch shortly then merge.
